### DISCUSSION TITLE
Add support for subsonic update

### DISCUSSION THREAD

**Discussion by FunkyWonder** [navidrome/navidrome#2752] [Ideas]
I would really like to externally update my navidrome library by using an endpoint. This way I can automatically update my library when beets finishes importing music.

**Comment by deluan** [comment]
3 ways to programatically start a scan:
- StartScan endpoint: https://opensubsonic.netlify.app/docs/endpoints/startscan/ (use this with beets' subsonic plugin)
- Send a SIGUSR1 to the Navidrome process: `pkill -USR1 navidrome`
- Call `navidrome scan` command:
```
$ navidrome scan --help
Scan music folder for updates

Usage:
  navidrome scan [flags]

Flags:
  -f, --full   check all subfolders, ignoring timestamps
  -h, --help   help for scan

Global Flags:
  -c, --configfile string    config file (default "./navidrome.toml")
      --datafolder string    folder to store application data (DB), needs write access (default ".")
  -l, --loglevel string      log level, possible values: error, info, debug, trace (default "info")
      --musicfolder string   folder where your music is stored (default "music")
  -n, --nobanner             don't show banner
```
